go/libraries/doltcore/dbfactory/file.go

```diff
@@ -192,6 +189,51 @@ func (fact FileFactory) CreateDB(ctx context.Context, nbf *types.NomsBinFormat,
 	return ddb, vrw, ns, nil
 }
 
+// CreateDB creates a local filesys backed database
+func (fact FileFactory) GetDBLoader(ctx context.Context, nbf *types.NomsBinFormat, urlObj *url.URL, params map[string]interface{}) (dbLoader DBLoader, err error) {
+	singletonLock.Lock()
+	defer singletonLock.Unlock()
+
+	path, err := url.PathUnescape(urlObj.Path)
+	if err != nil {
+		return nil, err
+	}
+
+	path = filepath.FromSlash(path)
+	path = urlObj.Host + path
+
+	err = validateDir(path)
+	if err != nil {
+		return nil, err
+	}
+
+	var useJournal bool
+	if params != nil {
+		_, useJournal = params[ChunkJournalParam]
+	}
+
+	var lock *fslock.Lock
+
+	if s, ok := singletonFileLocks[urlObj.Path]; ok {
+		lock = s
+	} else {
+		lock, err = nbs.AcquireManifestLock(path)
```

Comment:
These lifecycle changes seem weird and complicated and potentially wrong for non-journal stores (if those still exist...in that world the file manifest manager only hold the lock while it makes the switch...). Is there an alternative world where we change CreateDB() so that we can request ExeclusiveAccess and if it can't give it to us it can bail early and we can make lazy queryist use that instead?

Reply:
I guess we're not actually looking for Exclusive. We're looking for non-ReadOnly. Same thought though...